### DISCUSSION TITLE
Adding support for Ethers.js and Web3.js

### DISCUSSION THREAD

**Discussion by glitch-txs** [rainbow-me/rainbowkit#1316] [Ideas]
Hi team! 
As we know Ethers  v6 and Web3.js v4 have been recently released. And I wanted to present a project I've been working on called W3, it's a wallet connectors library that supports all ethereum libraries like ethers, web3.js, viem, etc.

I was wondering if there would be room for integrating it with RainbowKit's modal *(as an alternative and **not** as a replacement of wagmi's implementation)*, this would open the possibility for devs who don't use viem or wagmi to experience RainbowKit's solution with their own libraries of choice.

W3 has current support for all browser wallets, WalletConnect v2 and EIP-6963 compliant wallets.
You can learn more about it here: https://github.com/glitch-txs/w3
If I get some green light I could start working on the integration right away. Would love your thoughts!

Thanks!

**Comment by DanielSinclair** [comment]
@glitch-txs Very interesting project! Am open to you specing out what you think this might look like. Unfortunately I think the risk-reward may not line up for the amount of work required here, or the long-tail maintenance problems. The first problem I see is that RainbowKit is quite coupled with wagmi/viem hooks and types, which I imagine will continue to diverge from `ethers` over time. The 2nd problem would be around maintenance; the supply chain is already quite complex, and issues in viem or underlying connector libraries (i.e. walletconnect) will already bubble up through wagmi into RainbowKit dApps, and that would be dramatically exacerbated when more libraries and version diffs are introduced. As some of the first adopters of wagmi, it is still our firm belief that the wagmi way *is* the way.

**Reply by glitch-txs** [reply]
Understandable! Thanks for checking out! :)
